### PR TITLE
Security fix

### PR DESCRIPTION
**Pull request by Loriowar**

### CODE REVIEW

File: lib/auto_select2/select2_search_adapter/default.rb [hunk 1]
```diff
@@ -3,35 +3,54 @@ module Select2SearchAdapter
...
             get_init_values(
-                default_arel,
+                @searchable_class,
                 options[:item_ids],
                 options
             )
           end
         end
+
+        private
```
Comment (Loriowar) [inline lib/auto_select2/select2_search_adapter/default.rb]: maybe protected?

File: lib/generators/auto_select2/search_adapter_generator.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
+module AutoSelect2
+  module Generators
+    class SearchAdapterGenerator < Rails::Generators::NamedBase
+      source_root File.expand_path('../templates', __FILE__)
+
+      class_option :destination_path,
+                   type: :string,
+                   default: 'app/select2_search_adapters',
+                   desc: 'path where need to put generated class',
```
Comment (Loriowar) [inline lib/generators/auto_select2/search_adapter_generator.rb]: "Path for storing SearchAdapter classes"

File: lib/generators/auto_select2/search_adapter_generator.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                   banner: 'lib/search_adapters'
+      class_option :id_column,
+                   type: :string,
+                   desc: 'column of your model that will be sent to select2 as `id`',
```
Comment (Loriowar) [inline lib/generators/auto_select2/search_adapter_generator.rb]: Name of model column with identifier (i.e. `id`)

File: lib/generators/auto_select2/search_adapter_generator.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+                   banner: 'id',
+                   required: true
+      class_option :text_columns,
+                   type: :array,
+                   desc: 'columns of your model involved in search process',
```
Comment (Loriowar) [inline lib/generators/auto_select2/search_adapter_generator.rb]: Column for searching by

File: lib/generators/auto_select2/search_adapter_generator.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+                   banner: 'lastname firstname',
+                   required: true
+      class_option :hash_method,
+                   type: :string,
+                   desc: 'Method of your model that return a Hash that will be transported to select2',
```
Comment (Loriowar) [inline lib/generators/auto_select2/search_adapter_generator.rb]: Method of model for converting data into hash with select2 options (if present)

File: lib/generators/auto_select2/search_adapter_generator.rb [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+      end
+
+      private
+
+      def id_column
+        options[:id_column]
+      end
+
+      def text_columns
+        options[:text_columns].map { |c| ":#{c}" }
```
Comment (Loriowar) [inline lib/generators/auto_select2/search_adapter_generator.rb]: why so strange? maybe we need just `map(&:to_sym)`?

File: lib/auto_select2/select2_search_adapter/default.rb (hunk 1 above)
Comment (Tab10id) [inline lib/auto_select2/select2_search_adapter/default.rb]: For what?

File: lib/generators/auto_select2/search_adapter_generator.rb (hunk 6 above)
Comment (Tab10id) [inline lib/generators/auto_select2/search_adapter_generator.rb]: There is a generator.. text preparing

**Pull request merged**